### ISSUE TITLE
Segmentation fault with keras. regression problem? 2.1.6 works but 2.4.5 has segmentation fault.

### ISSUE THREAD

**Issue by boberyang2** [Nuitka/Nuitka#3063]
If you want to post a problem/bug, to help us understand and resolve your issue please check that
you have provided at least the information below, and discard up to here:

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version
2.4.5
Commercial: None
Python: 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)]
Flavor: CPython Official
Executable: D:\data_process\process_program_newest\venv\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10

- How did you install Nuitka and Python
pip

- The specific PyPI names and versions
keras

  > python -m pip list -v
Package                      Version  Location                                                      Installer
---------------------------- -------- ------------------------------------------------------------- ---------
absl-py                      2.1.0    D:\data_process\process_program_newest\venv\Lib\site-packages pip
astunparse                   1.6.3    D:\data_process\process_program_newest\venv\Lib\site-packages pip
certifi                      2024.7.4 D:\data_process\process_program_newest\venv\Lib\site-packages pip
charset-normalizer           3.3.2    D:\data_process\process_program_newest\venv\Lib\site-packages pip
flatbuffers                  24.3.25  D:\data_process\process_program_newest\venv\Lib\site-packages pip
gast                         0.6.0    D:\data_process\process_program_newest\venv\Lib\site-packages pip
google-pasta                 0.2.0    D:\data_process\process_program_newest\venv\Lib\site-packages pip
grpcio                       1.65.4   D:\data_process\process_program_newest\venv\Lib\site-packages pip
h5py                         3.11.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip
idna                         3.7      D:\data_process\process_program_newest\venv\Lib\site-packages pip
keras                        3.4.1    D:\data_process\process_program_newest\venv\Lib\site-packages pip
libclang                     18.1.1   D:\data_process\process_program_newest\venv\Lib\site-packages pip
Markdown                     3.6      D:\data_process\process_program_newest\venv\Lib\site-packages pip
markdown-it-py               3.0.0    D:\data_process\process_program_newest\venv\Lib\site-packages pip
MarkupSafe                   2.1.5    D:\data_process\process_program_newest\venv\Lib\site-packages pip
mdurl                        0.1.2    D:\data_process\process_program_newest\venv\Lib\site-packages pip
ml-dtypes                    0.4.0    D:\data_process\process_program_newest\venv\Lib\site-packages pip
namex                        0.0.8    D:\data_process\process_program_newest\venv\Lib\site-packages pip
Nuitka                       2.4.5    D:\data_process\process_program_newest\venv\Lib\site-packages pip
numpy                        1.26.4   D:\data_process\process_program_newest\venv\Lib\site-packages pip
opt-einsum                   3.3.0    D:\data_process\process_program_newest\venv\Lib\site-packages pip
optree                       0.12.1   D:\data_process\process_program_newest\venv\Lib\site-packages pip
ordered-set                  4.1.0    D:\data_process\process_program_newest\venv\Lib\site-packages pip
packaging                    24.1     D:\data_process\process_program_newest\venv\Lib\site-packages pip
pip                          22.3.1   D:\data_process\process_program_newest\venv\Lib\site-packages pip
protobuf                     4.25.4   D:\data_process\process_program_newest\venv\Lib\site-packages pip
Pygments                     2.18.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip
requests                     2.32.3   D:\data_process\process_program_newest\venv\Lib\site-packages pip
rich                         13.7.1   D:\data_process\process_program_newest\venv\Lib\site-packages pip
setuptools                   65.5.1   D:\data_process\process_program_newest\venv\Lib\site-packages pip
six                          1.16.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip
tensorboard                  2.17.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip
tensorboard-data-server      0.7.2    D:\data_process\process_program_newest\venv\Lib\site-packages pip
tensorflow                   2.17.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip
tensorflow-intel             2.17.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip
tensorflow-io-gcs-filesystem 0.31.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip
termcolor                    2.4.0    D:\data_process\process_program_newest\venv\Lib\site-packages pip
typing_extensions            4.12.2   D:\data_process\process_program_newest\venv\Lib\site-packages pip
urllib3                      2.2.2    D:\data_process\process_program_newest\venv\Lib\site-packages pip
Werkzeug                     3.0.3    D:\data_process\process_program_newest\venv\Lib\site-packages pip
wheel                        0.38.4   D:\data_process\process_program_newest\venv\Lib\site-packages pip
wrapt                        1.16.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip
zstandard                    0.23.0   D:\data_process\process_program_newest\venv\Lib\site-packages pip

- Many times when you get an error from Nuitka, your setup may be special
Nuitka: A segmentation fault has occurred. This is highly unusual and can
have multiple reasons. Please check https://nuitka.net/info/segfault.html
for solutions.

I CONFIRM THIS SEGMENTATION FALUT OCCURD WHEN I EXECUTE THE FOLLOWING CODES, AT THE MOMENT OF MODEL.FIT(...)
nuitka 2.1.6 work without problem, nuitka 2.3 and nuitka 2.4 with segmentation fault, i didnot try nuitka 2.2.

```
import os
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
import keras
from keras import layers
import numpy as np
from datetime import datetime as dt
from datetime import timedelta as td
import re
import sys
import shutil
import copy
import traceback
import pickle
import csv
if __name__ == '__main__':
    model = keras.Sequential(
        [
            layers.Input(shape=(500, 2)),
            layers.Conv1D(
                filters=64,
                kernel_size=7,
                padding="same",
                strides=2,
                activation="relu",
            ),
            layers.Dropout(rate=0.2),
            layers.Conv1D(
                filters=32,
                kernel_size=7,
                padding="same",
                strides=2,
                activation="relu",
            ),
            layers.Conv1DTranspose(
                filters=32,
                kernel_size=7,
                padding="same",
                strides=2,
                activation="relu",
            ),
            layers.Dropout(rate=0.2),
            layers.Conv1DTranspose(
                filters=64,
                kernel_size=7,
                padding="same",
                strides=2,
                activation="relu",
            ),
            layers.Conv1DTranspose(filters=4, kernel_size=7, padding="same"),
        ]
    )
    model.compile(optimizer=keras.optimizers.AdamW(learning_rate=0.001), loss="mse")
    model.summary()
    print('before fit!')
    input_d = np.zeros(shape=(700,500,2))
    output_d = np.zeros(shape=(700,500,4))
    a = model.fit(input_d,
                  output_d,
                  epochs=1, verbose=2,
                  batch_size=1,
                  validation_split=0.1,
                  callbacks=[
                      keras.callbacks.EarlyStopping(monitor="val_loss", patience=1, mode="min")
                  ], )
    print('after fit!')
    sys.exit()
```

- Provide in your issue the Nuitka options used
--standalone

- Note if this is a regression
nuitka 2.1.6 work without problem

**Comment by kayhayen** [comment]
I am not sure `keras 3.4.1` worked with Nuitka 2.1.6, but your example is great, and if it does, this would be a perfect case for git bisection, which is always good.

**Comment by kayhayen** [comment]
I managed to reproduce the segfault even with current 2.5 code, trying 2.1.6 with this setup right now.

**Comment by kayhayen** [comment]
My reproduction was on Linux, since it's easier to do there, but the 2.1.6 gave me an error, that I did resolve later I think, but maybe won't happen on Windows which is what you were using it seems, I will have to try that, but also I feel like maybe you are using new Nuitka and new keras vs. old Nuitka and older keras?

**Comment by kayhayen** [comment]
I am getting this error: https://github.com/Nuitka/Nuitka/issues/2862

That was not resolved until 2.4.x which one I don't know.

**Comment by kayhayen** [comment]
Probably Windows doesn't have as much of an issue distinguishing DLLs from extension modules, so I need to go there.

**Comment by kayhayen** [comment]
I bisected this to the introduction of tensorflow JIT support and fixed it in a rebase, and 2.5.5 hotfix contains it now.
